gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java

```diff
@@ -176,7 +189,7 @@ protected void registerPath(HiveSpec spec) throws IOException {
    * Or will create the table thru. RPC and return retVal from remote MetaStore.
    */
   private boolean ensureHiveTableExistenceBeforeAlternation(String tableName, String dbName, IMetaStoreClient client,
-      Table table, HiveSpec spec) throws TException{
+      Table table, HiveSpec spec) throws TException, IOException{
     try (AutoCloseableLock lock = this.locks.getTableLock(dbName, tableName)) {
```

Comment:
Isn't this lock a JVM-local lock? Won't we need Hive Metastore side locking to serialize updates?